File: 03842. Toggle Light Bulbs/solution.cs
/**
 * Problem: 3842. Toggle Light Bulbs
 *
 * Difficulty: Easy
 *
 * Language: C#
 *
 * Performance: Runtime - 0 ms (Beats 100%)
 */

public class Solution
{
  public IList<int> ToggleLightBulbs(IList<int> bulbs)
  {
    // Create a stack-allocated span for toggle states (1-100)
    Span<bool> toggleStates = stackalloc bool[101];

    // Toggle each bulb that appears in the input list
    for (int index = 0; index < bulbs.Count; index++)
    {
      // Get current bulb number
      int currentBulb = bulbs[index];

      // Flip the toggle state (on becomes off, off becomes on)
      toggleStates[currentBulb] = !toggleStates[currentBulb];
    }

    // Clear the original list to reuse it for results
    bulbs.Clear();

    // Collect bulbs that are currently on
    for (int i = 1; i < toggleStates.Length; i++)
      // If bulb is on, add it to the result list
      if (toggleStates[i])
        bulbs.Add(i);

    // Return the list of bulbs that are on
    return bulbs;
  }
}

File: 03842. Toggle Light Bulbs/solution.go
/**
 * Problem: 3842. Toggle Light Bulbs
 *
 * Difficulty: Easy
 *
 * Language: Golang
 *
 * Performance: Runtime - 0 ms (Beats 100%)
 */

func toggleLightBulbs(bulbs []int) []int {
	// Get number of bulbs to process
	n := len(bulbs)

	// Create boolean array to track toggle state (index 0 unused)
	freq := make([]bool, 101)

	// Toggle each bulb that appears in the array
	for i := 0; i < n; i++ {
		// If bulb is currently on, turn it off
		if freq[bulbs[i]] {
			freq[bulbs[i]] = false
		} else {
			// If bulb is currently off, turn it on
			freq[bulbs[i]] = true
		}
	}

	// Slice to store bulbs that are on
	ans := make([]int, 0)

	// Check all possible bulbs from 1 to 100
	for i := 0; i < 101; i++ {
		// If bulb is on, add to result
		if freq[i] {
			ans = append(ans, i)
		}
	}

	// Return the bulbs that are on
	return ans
}
